core/src/main/scala/org/apache/spark/deploy/worker/ExternalShuffleService.scala

```diff
@@ -64,3 +67,39 @@ class StandaloneWorkerShuffleService(sparkConf: SparkConf, securityManager: Secu
     }
   }
 }
+
+/**
+ * A main class for running the external shuffle service.
+ */
+object ExternalShuffleService extends Logging {
+  @volatile
+  private var server: ExternalShuffleService = _
+
+  private val barrier = new CountDownLatch(1)
+
+  def main(args: Array[String]): Unit = {
+    val sparkConf = new SparkConf
+    Utils.loadDefaultSparkProperties(sparkConf)
+    val securityManager = new SecurityManager(sparkConf)
+
+    // we override this value since this service is started from the command line
+    // and we assume the user really wants it to be running
+    sparkConf.set("spark.shuffle.service.enabled", "true")
```

Comment:
This is kind of weird. I would just add a `start` method in the shuffle service that ignores this flag. Really I think the flag should be checked outside of the shuffle service before instantiating an instance of it.